packages/client/src/reducers/contacts.js

```diff
@@ -0,0 +1,40 @@
+export const INITIAL_STATE = {
+  contacts: {},
+};
+
+const reducer = (state, action) => {
+  const address = Object.keys(action.payload)[0];
+  const body = action.payload[address];
+  const newContacts = (state.contacts?.[address] ?? []).slice(0);
+
+  switch (action.type) {
+    case "SET_CONTACT": {
+      const { index, newContact } = body;
+      newContacts[index] = newContact;

```

Comment:
using index to locate a contact is not reliable since they change all the time, suggest using the contact address
```suggestion
      const updateIndex = treasuryContacts.findIndex(c=>c.address === newContact.address);
      treasuryContacts[updateIndex] = newContact

```

Reply:
I went with index manipulation vs finding specific keys because the indexes only change when you add/remove people from contacts (there is no resorting currently) so it works fine for our purposes and felt like simpler code

Reply:
Got it, it makes sense. I'm thinking it's probably time for us to bring in immer for handling the state update, it can help make state update simple and reliable